frontend/lib/src/dataframes/arrowConcatUtils.ts

```diff
@@ -28,10 +28,38 @@ import {
   PandasColumnType,
   PandasIndexTypeName,
   PandasRangeIndex,
-  sameDataTypes,
-  sameIndexTypes,
 } from "./arrowTypeUtils"
 
+/** True if both arrays contain the same data types in the same order. */
+function sameDataTypes(
+  t1: PandasColumnType[],
+  t2: PandasColumnType[]
+): boolean {
+  // NOTE: We remove extra columns from the DataFrame that we add rows from.
```

Comment:
where do we remove this? This here is a standalone function, so the comment probably needs some more context or rephrasing. I would probably write it more generically: we consider Dataframes to have the same types if all columns that exist in `t1` also exist in `t2` in the same order and with the same type.

I would also extend the JSDoc above to state that t2 can be larger than t1 so that it is directly obvious from the doc string.

Reply:
This method was just moved out of `arrowTypeUtils` and it's untouched from me in its original form. While the overall `addRows` functionality would also require some refactoring, I do not plan to do it in the context of this project since the `add_rows` feature usage is almost non-existent (0.036%). It's in a ball range that we could consider deprecating/removing this feature all altogether.

Reply:
I cleaned up the comments based on your suggestion